client/mm/libxc/binance.go

```diff
@@ -0,0 +1,1194 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package libxc
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"decred.org/dcrdex/client/asset"
+	"decred.org/dcrdex/client/comms"
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/dex/calc"
+	"decred.org/dcrdex/dex/encode"
+)
+
+// Binance API spot trading docs:
+// https://binance-docs.github.io/apidocs/spot/en/#spot-account-trade
+
+const (
+	httpURL      = "https://api.binance.com"
+	websocketURL = "wss://stream.binance.com:9443"
+
+	usHttpURL      = "https://api.binance.us"
+	usWebsocketURL = "wss://stream.binance.us:9443"
+
+	testnetHttpURL      = "https://testnet.binance.vision"
+	testnetWebsocketURL = "wss://testnet.binance.vision"
```

Comment:
Testing on simnet, I see a lot of these
```
[ERR] MM[Arbitrage-127.0.0.1:17273-0-60]: Failed to subscribe to cex market: error initializing connection: websocketHandler remote connect: dial tcp: lookup testnet.binance.vision: i/o timeout
```
which tanks the arb bot altogether from what I can tell, since we abandon reconnect handling. Is there a timeout value we can adjust somewhere to avoid this?

Reply:
I've never seen it.. and according to this https://pkg.go.dev/net#Dialer:

```
	// The default is no timeout.
```

Are you sure there's no issue on your end? Also, Binance testnet may have been having some issues. Earlier I was very sporadically getting messages on the orderbook stream, but now it's coming through nicely.

Reply:
Nevermind. This was my problem.